test/integration/targets/dnf/tasks/repo.yml

```diff
@@ -560,42 +535,43 @@
   always:
     - name: Clean up
       dnf:
-        name: dinginessentail
+        name: multilib-dinginessentail
         state: absent
+  when: ansible_architecture == "x86_64"
 
 - block:
-  - name: make sure dinginessentail is not installed
+  - name: make sure multilib-dinginessentail is not installed
     dnf:
-      name: dinginessentail
+      name: multilib-dinginessentail
       state: absent
 
-  - name: install dinginessentail both archs
+  - name: install multilib-dinginessentail both archs
     dnf:
       name:
-        - "{{ repodir }}/dinginessentail-1.1-1.x86_64.rpm"
-        - "{{ repodir_i686 }}/dinginessentail-1.1-1.i686.rpm"
+        - "{{ repodir }}/multilib-dinginessentail-1.1-1.x86_64.rpm"
+        - "{{ repodir }}/multilib-dinginessentail-1.1-1.i686.rpm"
       state: present
       disable_gpg_check: true
 
-  - name: try to install lower version of dinginessentail from rpm file, without allow_downgrade, just one arch
+  - name: try to install lower version of multilib-dinginessentail from rpm file, without allow_downgrade, just one arch
     dnf:
-      name: "{{ repodir_i686 }}/dinginessentail-1.0-1.i686.rpm"
+      name: "{{ repodir }}/multilib-dinginessentail-1.0-1.i686.rpm"
       state: present
     register: dnf_result
 
-  - name: check dinginessentail with rpm
-    shell: rpm -q dinginessentail
+  - name: check multilib-dinginessentail with rpm
+    shell: rpm -q multilib-dinginessentail
     register: rpm_result
 
   - name: verify installation
     assert:
       that:
           - "not dnf_result.changed"
-          - "rpm_result.stdout_lines[0].startswith('dinginessentail-1.1-1')"
-          - "rpm_result.stdout_lines[1].startswith('dinginessentail-1.1-1')"
+          - "rpm_result.stdout_lines[0].startswith('multilib-dinginessentail-1.1-1')"
+          - "rpm_result.stdout_lines[1].startswith('multilib-dinginessentail-1.1-1')"
   always:
     - name: Clean up
       dnf:
-        name: dinginessentail
+        name: multilib-dinginessentail
         state: absent
   when: ansible_architecture == "x86_64"
```

Comment:
I realize this condition isn't new, but since the block is being updated, I have the same question here. Can this be adapted to also work when the tests are run using aarch64?

Reply:
I think I found a way, see https://github.com/ansible/ansible/pull/83624/commits/6c3a7d1dc5b364fc302bf444c8126e04bca48f3b. Works on my M2 Pro machine ;-)